openedx/features/enterprise_support/utils.py

```diff
@@ -329,17 +329,22 @@ def get_enterprise_learner_portal(request):
     # Only cache this if a learner is authenticated (AnonymousUser exists and should not be tracked)
 
     learner_portal_session_key = 'enterprise_learner_portal'
-
+    enterprise_customer_session_key = 'enterprise_customer'
     if enterprise_enabled() and ENTERPRISE_HEADER_LINKS.is_enabled() and user and user.id:
         # If the key exists return that value
-        if learner_portal_session_key in request.session:
-            return json.loads(request.session[learner_portal_session_key])
+        if learner_portal_session_key in request.session and enterprise_customer_session_key in request.session:
+            learner_portal_session = json.loads(request.session[learner_portal_session_key])
+            if request.session[enterprise_customer_session_key].get('slug') == learner_portal_session.get('slug'):
+                return learner_portal_session
```

Comment:
what should be returned when both slug values are different?

Reply:
In that case first priority is given to the request.session[enterprise_customer_session_key] as this is the value which is set at the time when active enterprise is changed. And the second priority is given to the enterprise which is in the request.